kafka-cron/README.md

```diff
@@ -0,0 +1,162 @@
+
+The `cron` tool common to Unix operating systems runs jobs on a schedule. Cron only works on a single hosts. We want to create a version of cron that can schedule jobs across multiple workers, running on different hosts.
+
+### Background on Apache Kafka
+
+Kafka is an open-source distributed queue. You can read about the core Kafka concepts in the [Kafka: a Distributed Messaging System for Log Processing paper](https://www.microsoft.com/en-us/research/wp-content/uploads/2017/09/Kafka.pdf).
+
+After reading that paper you should understand:
+ * How Kafka stores data
+ * What producers, consumers, and brokers are
+ * What topics and partitions are
+
+### Part 1: Distributed cron with one queue
+
+In this section of the project, we will start by creating a functional distributed cron system. We will build two separate programs:
+ * A Kafka producer that reads configuration files for jobs and queues jobs for execution
+ * A Kafka consumer that dequeues jobs from a queue and runs them
+
+We'll need to be able to run Kafka. The easiest way is to use `docker-compose`. The [conduktor/kafka-stack-docker-compose](https://github.com/conduktor/kafka-stack-docker-compose) project provides several starter configurations for running Kafka. The config for `zk-single-kafka-single.yml` will work for development purposes.
+
+There is a [Golang Kafka client](https://docs.confluent.io/kafka-clients/go/current/overview.html#go-example-code) that we will use to interact with Kafka.
+
+We may want to run other Docker containers later, so we may want to make our own copy of that configuration that we can add to. 
+
+Our producer program needs to be able to do the following:
+ * Read and parse a file with cron job definitions (we'll set up our own for this project, don't reuse the system cron config file because we will want to modify the format later)
+ * Write a message to Kafka specifying the command to run, the intended start time of the job, and any other information that we think is necessary. It probably makes sense to encode this information as JSON (see [Go By Example: JSON](https://gobyexample.com/json) if you have never worked with JSON in Golang before)
+ * We will also need to [create a Kafka topic](https://kafka.apache.org/documentation/#quickstart_createtopic). In a production environment we would probably use separate tooling to manage topics (perhaps Terraform), but for this project, we can [create our Kafka topic using code](https://github.com/confluentinc/examples/blob/7.3.0-post/clients/cloud/go/producer.go#L39).
+
+Our consumer program needs to be able to do the following things:
+ * Read job information from a Kafka queue (decoding JSON)
+ * Execute the commands to run the jobs (assume this is a simple one-line command that you can `exec` for now)
+ * Because the consumer is writing jobs to the queue when they are ready to run, your consumer does not need to do any scheduling or to parse crontab format
```

Comment:
```suggestion
 * Because the producer is writing jobs to the queue when they are ready to run, your consumer does not need to do any scheduling or to parse crontab format
```